app/screens/convert_gm_to_channel/message_box/message_box.tsx

```diff
@@ -0,0 +1,141 @@
+import {changeOpacity, makeStyleSheetFromTheme} from '@app/utils/theme';
+import {typography} from '@app/utils/typography';
+
+type MessageBoxTypes = 'default' | 'danger'
+
+type Props = {
+    header: string;
+    body: string;
+    type?: MessageBoxTypes;
+}
+
+const getBaseStyles = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            backgroundColor: changeOpacity(theme.sidebarTextActiveBorder, 0.08),
+            borderWidth: 1,
+            borderRadius: 8,
+            borderColor: changeOpacity(theme.sidebarTextActiveBorder, 0.16),
+            display: 'flex',
+            flexDirection: 'row',
+            padding: 16,
+            gap: 12,
+        },
+        icon: {
+            marginTop: 5,
+            fontSize: 20,
+            width: 28,
+            height: 28,
+            borderWidth: 3,
+            color: theme.sidebarTextActiveBorder,
+            borderColor: theme.sidebarTextActiveBorder,
+            borderRadius: 14,
+            textAlign: 'center',
+        },
+        iconContainer: {},
+        textContainer: {
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 8,
+            flex: 1,
+        },
+        heading: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'SemiBold'),
+        },
+        body: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100),
+        },
+    };
+});
+
+const getDefaultStylesFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            backgroundColor: changeOpacity(theme.sidebarTextActiveBorder, 0.08),
+            borderColor: changeOpacity(theme.sidebarTextActiveBorder, 0.16),
+        },
+        icon: {
+            color: theme.sidebarTextActiveBorder,
+            borderColor: theme.sidebarTextActiveBorder,
+        },
+    };
+});
+
+const getDangerStylesFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            backgroundColor: changeOpacity(theme.dndIndicator, 0.08),
+            borderColor: changeOpacity(theme.dndIndicator, 0.16),
+        },
+        icon: {
+            color: theme.dndIndicator,
+            borderColor: theme.dndIndicator,
+        },
+    };
+});
+
+const getStyleFromTheme = (theme: Theme, kind: MessageBoxTypes | undefined) => {
+    const baseStyles = getBaseStyles(theme);
+
+    let kindStyles;
+    switch (kind) {
+        case 'danger': {
+            kindStyles = getDangerStylesFromTheme(theme);
+            break;
+        }
+        default: {
+            kindStyles = getDefaultStylesFromTheme(theme);
+            break;
+        }
+    }
+
+    return {
+        ...baseStyles,
+        ...kindStyles,
+    };
+};
+
+const MessageBox = ({
+    header,
+    body,
+    type,
+}: Props) => {
+    const theme = useTheme();
+    const styles = getBaseStyles(theme);
+    const foo = getStyleFromTheme(theme, type);
```

Comment:
what is foo?

Reply:
A very classy variable name 😎 just kidding, fixed it now.